src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1669,6 +1674,11 @@ private void emitNoResolvedType(NoType type, String defaultEmit) {
         return;
       }
       String displayName = type.getDisplayName();
+      // In partial mode, closure doesn't know the correct name of imported symbols, if the name
+      // matches one in the precomputed map, replace it with the original declared name
+      if (importRenameMap.containsKey(displayName)) {
```

Comment:
noResolvedType can also happen in non-incremental mode (if code is incorrect/missing deps), which would then cause an `NPE` here. Maybe always initialize the map?

Reply:
Ah I see. That's a bit surprising that you assign the field twice, in two different locations. I guess the alternative (passing it into `ImportRenameMapBuilder`) isn't worth it.

Can you use `Collections.emptyMap()` up there to make it a bit more obvious that it is never used (and prevent accidental use)?

Reply:
Done!